src/ripple/rpc/handlers/Peers.cpp

```diff
@@ -32,17 +32,31 @@ doPeers(RPC::JsonContext& context)
 {
     Json::Value jvResult(Json::objectValue);
 
+    jvResult[jss::peers] = context.app.overlay().json();
+
+    // Legacy support
+    if (context.apiVersion == 1)
     {
-        jvResult[jss::peers] = context.app.overlay().json();
+        for (auto& p : jvResult[jss::peers])
+        {
+            if (auto s = p[jss::track].asString(); !s.empty())
```

Comment:
This code has a subtle bug courtesy of our JSON library.  By _examining_ `jss::track` in this way you have injected `jss::track` into object `s` with a value of `null`.  You can avoid the bug with brute force by doing something like this:
```
        for (auto& p : jvResult[jss::peers])
        {
            if (p.isMember(jss::track)) // Important test
            {
                if (auto s = p[jss::track].asString(); !s.empty())
                {
                    if (s == "diverged")
                        p["sanity"] = "insane";
                    if (s == "unknown")
                        p["sanity"] = "unknown";
                }
            }
        }
```
Or perhaps you have a more clever suggestion.

Reply:
Ooof, you're right. I totally missed that, thanks. I'll adopt your approach.

Reply:
I've fixed this thusly: 

https://github.com/ripple/rippled/blob/c52a3db8db49214385ab9f3f1d2944a2412e4222/src/ripple/rpc/handlers/Peers.cpp#L38-L52